server/src/com/thoughtworks/go/server/service/support/ThreadInformationProvider.java

```diff
@@ -68,15 +77,21 @@ public double priority() {
         return json;
     }
 
-    private TreeMap<Long, Map<String, Object>> getStackTraceInformation(ThreadMXBean threadMXBean) {
+    private TreeMap<Long, Map<String, Object>> getThreadInformation(ThreadMXBean threadMXBean) {
         TreeMap<Long, Map<String, Object>> traces = new TreeMap<>();
         ThreadInfo[] threadInfos = threadMXBean.dumpAllThreads(true, true);
         for (ThreadInfo threadInfo : threadInfos) {
             LinkedHashMap<String, Object> threadStackTrace = new LinkedHashMap<>();
             threadStackTrace.put("Id", threadInfo.getThreadId());
             threadStackTrace.put("Name", threadInfo.getThreadName());
             threadStackTrace.put("State", threadInfo.getThreadState());
-
+            threadStackTrace.put("UserTime(nanoseconds)", threadMXBean.getThreadUserTime(threadInfo.getThreadId()));
+            threadStackTrace.put("CPUTime(nanoseconds)", threadMXBean.getThreadCpuTime(threadInfo.getThreadId()));
+            threadStackTrace.put("DaemonThreadInfo", daemonThreadStatsCollector.statsFor(threadInfo.getThreadId()));
+            if (threadMXBean instanceof com.sun.management.ThreadMXBean) {
```

Comment:
Checking for type on a JVM that does not have the required class will throw a CNFE when loading `ThreadInformationProvider` class, which is why I suggested the duck-type approach.